med-inventory/src/components/purchase-medicine/purchase-layout/purchaseLayout.js

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from "react";
+import Card from "@material-ui/core/Card";
+import axios from "axios";
+import CardContent from "@material-ui/core/CardContent";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import { AgGridReact } from "ag-grid-react";
+import "ag-grid-community/dist/styles/ag-grid.css";
+import "ag-grid-community/dist/styles/ag-theme-balham.css";
+import "./purchaseLayout.css";
+import NumericEditor from "./numericEditor.jsx";
+import PreviewPurchase from "./previewPurchase";
+
+class purchaseLayout extends Component{
+  constructor(){
+    super();
+    this.gridApi={};
+    this.gridOptions = {
+      suppressCellSelection: true,
+      suppressPropertyNamesCheck: true,
+      suppressRowClickSelection: true,
+      accentedSort: true,
+      enableBrowserTooltips: true,
+      suppressPreventDefaultOnMouseWheel: true,
+      suppressDragLeaveHidesColumns: true,
+      headerHeight: 46
+    }
+
+    this.state = {
+      columnDefs: [
+        {
+          headerName: "NAME", field: "name" ,sortable: true, filter: true ,pinned:"left", width:180,
+          headerCheckboxSelection: true, checkboxSelection: true, lockPosition: true
+        }, {
+          headerName: "MANUFACTURER", field: "manufacturer",sortable: true,  width:150, filter: true
+        }, {
+          headerName: "PRICE", field: "mrp",sortable: true,  width:150, filter: true
+        },{
+          headerName : "CREATED BY", field :"CreatedBy",sortable: true, width:150, filter: true
+        },{
+          headerName : "CATEGORY" , field : "category" ,sortable : true , width:150, filter :true
+        },{
+          headerName : "CURRENT STOCK COUNT" , resizable :"true",field : "currentStockCount" , width:180, sortable : true , filter : true
+        },{
+          headerName : "NEW STOCK COUNT" , field : "newStockCount" , sortable : true , width:180, filter : true,
+          editable: true,cellEditor: "numericEditor",pinned: "right", lockPinned: true
+        }
+      ],
+      rowData: [],
+      rowSelection : "multiple",
+      singleEditClick : "true",
```

Comment:
I don't see this state value being changed anywhere, do we need this as part of state?

Reply:
Why is this resolved?